lib/internal/worker.js

```diff
@@ -79,12 +81,32 @@ class Worker extends EventEmitter {
       }
     }
 
+    let env;
+    if (typeof options.env === 'object' && options.env !== null) {
+      env = Object.create(null);
+      for (const [ key, value ] of Object.entries(options.env))
+        env[key] = `${value}`;
+    } else if (options.env == null) {
+      env = process.env;
+    } else if (options.env !== shareEnv) {
+      throw new ERR_INVALID_ARG_TYPE('options.env',
+                                     ['object', 'undefined', 'null'],
```

Comment:
From the doc above, `shareEnv` is
> A special value that can be passed as the `env` option of the [`Worker`][]
constructor, to indicate that the current thread and the Worker thread should
share environment variables.

Maybe `shareEnv` should be added here ?

(So we let `env` to be `undefined` here on purpose if `options.env === shareEnv` ?)

Reply:
@ZYSzys Yup, added `shareEnv`, thank you.

> (So we let `env` to be `undefined` here on purpose if `options.env === shareEnv` ?)

Yes, that’s on purpose.